src/poetry/utils/authenticator.py

```diff
@@ -202,18 +194,17 @@ def request(
         session = self.get_session(url=url)
         prepared_request = session.prepare_request(request)
 
-        proxies = kwargs.get("proxies", {})
-        stream = kwargs.get("stream")
+        proxies: dict[str, str] = kwargs.get("proxies", {})
+        stream: bool | None = kwargs.get("stream")
 
         certs = self.get_certs_for_url(url)
-        verify = kwargs.get("verify") or certs.cert or certs.verify
-        cert = kwargs.get("cert") or certs.client_cert
+        verify: bool | str | Path = kwargs.get("verify") or certs.cert or certs.verify
+        verify = str(verify) if isinstance(verify, Path) else verify
 
+        cert: str | Path | None = kwargs.get("cert") or certs.client_cert
```

Comment:
```suggestion
        cert: str | Path | None = kwargs.get("cert", certs.client_cert)
```
Isn't this the same?

Reply:
not _quite_ the same, if `cert` is somehow an empty string...

Reply:
Oh, right, didn't think about this case.